WorkloadManagementSystem/Service/JobStateUpdateHandler.py

```diff
@@ -183,61 +181,51 @@ def export_setJobStatusBulk(self, jobID, statusDict):
     lastTime = max([float(t) for s, t in result['Value'].items() if s != 'LastTime'])
     lastTime = Time.toString(Time.fromEpoch(lastTime))
 
-    # Get the last status values
     dates = sorted(statusDict)
     # We should only update the status if its time stamp is more recent than the last update
-    for date in [date for date in dates if date >= lastTime]:
-      sDict = statusDict[date]
-      if sDict['Status']:
-        status = sDict['Status']
+    if dates[-1] >= lastTime:
+      # Get the last status values
+      for date in [date for date in dates if date >= lastTime]:
+        sDict = statusDict[date]
+        status = sDict.get('Status', status)
         if status in JobStatus.JOB_FINAL_STATES:
           endDate = date
-        if status == JobStatus.RUNNING:
-          startFlag = JobStatus.RUNNING
-      if sDict['MinorStatus']:
-        minor = sDict['MinorStatus']
-        if minor == "Application" and startFlag == JobStatus.RUNNING:
+        minor = sDict.get('MinorStatus', minor)
+        # Pick up the start date
+        if minor == "Application" and status == JobStatus.RUNNING and not startDate:
           startDate = date
-      if sDict['ApplicationStatus']:
-        application = sDict['ApplicationStatus']
-      counter = sDict.get('ApplicationCounter')
-      if counter:
-        appCounter = counter
-    attrNames = []
-    attrValues = []
-    if status:
-      attrNames.append('Status')
-      attrValues.append(status)
-    if minor:
-      attrNames.append('MinorStatus')
-      attrValues.append(minor)
-    if application:
-      attrNames.append('ApplicationStatus')
-      attrValues.append(application)
-    if appCounter:
-      attrNames.append('ApplicationCounter')
-      attrValues.append(appCounter)
-    result = jobDB.setJobAttributes(jobID, attrNames, attrValues, update=True)
-    if not result['OK']:
-      return result
+        application = sDict.get('ApplicationStatus', application)
+        appCounter = sDict.get('ApplicationCounter', appCounter)
+
+      attrNames = []
+      attrValues = []
+      if status:
+        attrNames.append('Status')
+        attrValues.append(status)
+      if minor:
+        attrNames.append('MinorStatus')
+        attrValues.append(minor)
+      if application:
+        attrNames.append('ApplicationStatus')
+        attrValues.append(application)
+      if appCounter:
+        attrNames.append('ApplicationCounter')
+        attrValues.append(appCounter)
+      result = jobDB.setJobAttributes(jobID, attrNames, attrValues, update=True)
+      if not result['OK']:
+        return result
 
-    if endDate:
-      result = jobDB.setEndExecTime(jobID, endDate)
-    if startDate:
-      result = jobDB.setStartExecTime(jobID, startDate)
+      if endDate:
+        result = jobDB.setEndExecTime(jobID, endDate)
+      if startDate:
+        result = jobDB.setStartExecTime(jobID, startDate)
```

Comment:
Why did you indent these 4 last lines?

Reply:
Will look, I didn't do anything, it's my editor that did something ;-)

Reply:
Oups, no. it was me... looking at why!

Reply:
Ah yes, we change the jobDB information only if there was an update more recent than the last one (i.e. we don't touch it if this is coming via a request), but you may be right... I have to think a bit about it ;-)... If we need to update it it is really messy...

Reply:
It should be better now... The problem is that this is very hard to test as it only happens if state is updated in bulk by a request!